cocotb/_py_compat.py

```diff
@@ -133,15 +51,11 @@ def __enter__(self):
     def __exit__(self, *excinfo):
         pass
 
-
-# https://stackoverflow.com/a/38668373
-# Backport of abc.ABC, compatible with Python 2 and 3
-abc_ABC = abc.ABCMeta('ABC', (object,), {'__slots__': ()})
-
-
-# source TimeoutError introduced in Python 3.3 to be used by timeout functions
-if sys.version_info < (3, 3):
-    class TimeoutError(OSError):
-        pass
+# On python 3.7 onwards, `dict` is guaranteed to preserve insertion order.
+# Since `OrderedDict` is a little slower that `dict`, we prefer the latter
+# when possible.
+if sys.version_info[:2] >= (3, 7):
+    insertion_ordered_dict = dict
 else:
-    TimeoutError = TimeoutError
+    import collections
+    insertion_ordered_dict = collections.OrderedDict
```

Comment:
Can you double-check if OrderedDict isn't making that optimization itself already? (I know you copied this code around, but let's not copy things around which are outdated.)

Reply:
It's not, because `OrderedDict` has extra features to change the order that `dict` does not have. We don't need those features here.